khi_robot_control/CMakeLists.txt

```
@@ -20,36 +32,35 @@ include_directories(
 
 ## KRNX
 ## TODO: external library path
-link_directories(${CMAKE_CURRENT_SOURCE_DIR}/lib)
 if( ${CMAKE_SYSTEM_PROCESSOR} MATCHES ".*x86_64.*" )
-  set(krnx_LIBRARY krnx-x86_64)
+  set(krnx_lib_name krnx-x86_64)
 elseif( ${CMAKE_SYSTEM_PROCESSOR} MATCHES ".*aarch64.*" )
-  set(krnx_LIBRARY krnx-aarch64)
+  set(krnx_lib_name krnx-aarch64)
 elseif( ${CMAKE_SYSTEM_PROCESSOR} MATCHES ".*i386.*" )
-  set(krnx_LIBRARY krnx-i386)
+  set(krnx_lib_name krnx-i386)
 elseif( ${CMAKE_SYSTEM_PROCESSOR} MATCHES ".*armhf.*" )
-  set(krnx_LIBRARY krnx-armhf)
+  set(krnx_lib_name krnx-armhf)
 elseif( ${CMAKE_SYSTEM_PROCESSOR} MATCHES ".*armel.*" )
-  set(krnx_LIBRARY krnx-armel)
+  set(krnx_lib_name krnx-armel)
 else()
-  set(krnx_LIBRARY krnx-x86_64)
+  message(WARNING "\nCould not detect system processor architecture: assuming 'x86_64'. This may result in non-working binaries.")
+  set(krnx_lib_name krnx-x86_64)
 endif()
 
+add_library(krnx_lib SHARED IMPORTED)
+set_target_properties(krnx_lib PROPERTIES
+  IMPORTED_LOCATION ${CMAKE_CURRENT_LIST_DIR}/lib/lib${krnx_lib_name}.so)
+set(krnx_LIBRARIES krnx_lib)
+
 ## Declare a C++ library
-add_library(khi_robot_client src/khi_robot_krnx_driver.cpp src/khi_robot_client.cpp )
+add_library(khi_robot_client src/khi_robot_client.cpp src/khi_robot_krnx_driver.cpp)
 
```

I executed `catkin_make` in this revision, and I got include error `khi_robot_msgs/KhiRobotCmd.h`.
Maybe it needs `add_dependencies` about srv.

```suggestion
add_dependencies(khi_robot_client ${${PROJECT_NAME}_EXPORTED_TARGETS} ${catkin_EXPORTED_TARGETS})
```

---

I've dropped the commit that introduced this change.

I only build with `catkin_tools`, and that seems to order targets slightly differently, so I did not run into this.